Gems/EMotionFX/Code/Source/Integration/Editor/Components/EditorActorComponent.cpp

```diff
@@ -544,8 +544,57 @@ namespace EMotionFX
 
         void EditorActorComponent::OnAssetReloaded(AZ::Data::Asset<AZ::Data::AssetData> asset)
         {
+            // Release the asset so everything can get unloaded.
+            // The Actor asset holds a reference to a ModelAsset which can only be reloaded with a manual call.
+            // Since the Actor asset passed into this function has already been reloaded with the old ModelAsset,
+            // let it and the current Actor reference unload first.
+            // In the Unloaded event, the model will be requested for reload.
+            // When the model has finished reloading, the Actor will be QueueLoaded and will pick up the newly reloaded ModelAsset.
+            m_reloading = true;
             DestroyActorInstance();
-            OnAssetReady(asset);
+            m_actorAsset.Release();
+        }
+
+        void EditorActorComponent::OnAssetUnloaded(AZ::Data::AssetId assetId, AZ::Data::AssetType)
+        {
+            if(!m_reloading)
+            {
+                return;
+            }
+
+            m_reloading = false;
```

Comment:
setting m_reloading to false here means that this code will not run the next time OnAssetUnloaded() is called... if there is an error at link 575

So is it an error if m_reloading is false when starting this code?

Reply:
The asset might be unloaded for other reasons, so I wouldn't want to error in that case.  If the asset is already unloaded, it can't unload again until after OnAssetLoad is called, in which case OnAssetReload should get called again when an update happens, so everything will just reset itself in that case.